### DISCUSSION TITLE
How does I Run migration at application startup

### DISCUSSION THREAD

**Discussion by sontixyou** [SeaQL/sea-orm#2311] [Q&A]
## Question

I try to run migration at application startup when I reading [this document](https://www.sea-ql.org/sea-orm-cookbook/007-run-migration-at-app-startup.html).
sea-orm: ver1.0.0

```rust
use sea_orm::{ConnectionTrait, Database, DatabaseConnection};
use migration::{Migrator, MigratorTrait};

let connection = Database::connect(&database_origin_url).await?;
Migrator::up(&connection, None).await.unwrap();
```

I am having trouble performing the migration because of this error.

```rust

error[E0277]: the trait bound `&sea_orm::DatabaseConnection: migration::IntoSchemaManagerConnection<'_>` is not satisfied
   --> src/database.rs:38:18
    |
38  |     Migrator::up(&connection, None).await.unwrap();
    |     ------------ ^^^^^^^^^^^ the trait `migration::IntoSchemaManagerConnection<'_>` is not implemented for `&sea_orm::DatabaseConnection`
    |     |
    |     required by a bound introduced by this call
    |
    = help: the following other types implement trait `migration::IntoSchemaManagerConnection<'c>`:
              &'c migration::sea_orm::DatabaseConnection
              &'c migration::sea_orm::DatabaseTransaction
              migration::SchemaManagerConnection<'c>
note: required by a bound in `migration::MigratorTrait::up`
   --> /Users/user_name/.cargo/registry/src/index.crates.io-6f17d22bba15001f/sea-orm-migration-1.0.0/src/migrator.rs:231:12
    |
229 |     async fn up<'c, C>(db: C, steps: Option<u32>) -> Result<(), DbErr>
    |              -- required by a bound in this associated function
230 |     where
231 |         C: IntoSchemaManagerConnection<'c>,
    |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `MigratorTrait::up`
```

Can I just implement it myself to avoid the error?

**Comment by Shaphil** [comment]
I ran into the same problem while trying to run an Actix-Web application after integrating Sea-ORM,

```cmd
error[E0277]: the trait bound `&sea_orm::DatabaseConnection: IntoSchemaManagerConnection<'_>` is not satisfied
   --> src/main.rs:21:18
    |
21  |     Migrator::up(&db, None).await.unwrap();
    |     ------------ ^^^ the trait `IntoSchemaManagerConnection<'_>` is not implemented for `&sea_orm::DatabaseConnection`
    |     |
    |     required by a bound introduced by this call
    |
    = help: the following other types implement trait `IntoSchemaManagerConnection<'c>`:
              &'c migration::sea_orm::DatabaseConnection
              &'c migration::sea_orm::DatabaseTransaction
              SchemaManagerConnection<'c>
note: required by a bound in `migration::MigratorTrait::up`
   --> C:\Users\Shaphil\.cargo\registry\src\index.crates.io-6f17d22bba15001f\sea-orm-migration-0.12.15\src\migrator.rs:230:12
    |
228 |     async fn up<'c, C>(db: C, steps: Option<u32>) -> Result<(), DbErr>
    |              -- required by a bound in this associated function
229 |     where
230 |         C: IntoSchemaManagerConnection<'c>,
    |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `MigratorTrait::up`

For more information about this error, try `rustc --explain E0277`.
error: could not compile `rust-actix` (bin "rust-actix") due to 1 previous error
```

This is the line that produces the error,

```rust
Migrator::up(&db, None).await.unwrap();
```

The application runs fine if I comment out that line.

I am also using `v1.0.0`,

```toml
sea-orm = { version = "1.0.0", features = ["sqlx-postgres", "runtime-tokio-rustls", "macros"] }
```

Here is the `main.rs` file - [https://github.com/koddlers/rust-actix/blob/master/src/main.rs](https://github.com/koddlers/rust-actix/blob/master/src/main.rs)

I was following along a tutorial on Youtube, [https://www.youtube.com/watch?v=ENgFBKTBqDU&list=PLGOIZXklfFkRh8jHNY8070KUl86Tj3Ztf](https://www.youtube.com/watch?v=ENgFBKTBqDU&list=PLGOIZXklfFkRh8jHNY8070KUl86Tj3Ztf)

and, this guide for integration, [https://www.sea-ql.org/SeaORM/docs/migration/setting-up-migration/#app-crate](https://www.sea-ql.org/SeaORM/docs/migration/setting-up-migration/#app-crate)

What am I doing wrong, and how can I fix this?

**Reply by Shaphil** [reply]
Solved! Downgrading to `v0.12.15` worked for me.

In `cargo.toml` I changed from,
```toml
sea-orm = { version = "1.0.0", features = ["sqlx-postgres", "runtime-tokio-rustls", "macros"] }
```

to
```toml
sea-orm = { version = "0.12.15", features = ["sqlx-postgres", "runtime-tokio-rustls", "macros"] }
```

and the application is now running perfectly. the discussion from sea-orm discord helped a lot.

@sontixyou, check this out.

**Reply by sontixyou** [reply]
Try it. Thanks.
But I want to move up sea-orm to v1.0.0 so I can get Migratior working properly!

**Reply by Shaphil** [reply]
I actually, got this solution from Sea-ORM Discord. The suggested version was something like `v0.11.x` but I tried `v0.12.15` and it still worked fine.

I think maybe in `v1.0.0` the migrator setup may change which hasn't been reflected in the official documentation yet. We'll have to wait and see. But for now, rolling back to a previous version (again, for me it's `v0.12.15`) works fine as a workaround.

**Comment by sontixyou** [comment]
https://github.com/SeaQL/sea-orm/issues/2341
